samples/python/thermo/vapordome.py

```diff
@@ -7,10 +7,14 @@
 Keywords: thermodynamics, non-ideal fluid, plotting
 """
 
-import cantera as ct
-import pandas as pd
 import numpy as np
-from matplotlib import pyplot as plt
+try:
+    import pandas as pd
+except ImportError:
+    print("Example requires pandas installation")
+    exit(0)
```

Comment:
I'm not a big fan of this pattern. The error message and resolution is already obvious from the import error, and this makes the happy path example code a little harder to follow for new Python people, especially if they're coming from Matlab. For optional dependencies, it _may_ make sense but even then, I'm skeptical. YMMV ☺️

Reply:
Do you have a better suggestion? These patterns are widely used elsewhere in the Python sample folders. The only thing I did differently here was to add the `exit(0)` so I don't have to indent everything.

Fwiw, this came up when running your nifty:
```
          find samples/python -type f -iname "*.py" \
          -exec sh -c 'for n; do echo "$n" | tee -a results.txt && python3 "$n" >> results.txt || exit 1; done' sh {} +
```
locally, which crashes out for this example when `pandas` is not installed.

Reply:
That's kinda the point, isn't it? For it to fail when dependencies aren't installed, so we know one of the samples isn't being run/tested? I know this is used elsewhere in the examples but IMO it's an anti-pattern for example code. I think it's probably a holdover from when it was difficult to install dependencies, especially on Windows, but that's been more-or-less solved for a few years now. I don't think it's a huge deal, but I don't think it actually helps users, and it may be harmful insofar as it allows the CI to pass when it really shouldn't

Reply:
:+1: ... I see your point, and dropped the commit.